pyiron_atomistics/sphinx/output_parser.py

```diff
@@ -386,3 +387,97 @@ def get_convergence(self):
     def get_fermi(self):
         pattern = r"Fermi energy:\s+(\d+\.\d+)\s+eV"
         return np.array(re.findall(pattern, self.log_main)).astype(float)
+
+
+class SphinxWavesParser:
+    """Class to read SPHInX waves.sxb files (HDF5 format)
+
+    Initialize with waves.sxb filename, or use load ()
+
+    """
+
+    def __init__(self, file_name="waves.sxb", cwd="."):
+        """
+        Args:
+            file_name (str): file name
+            cwd (str): directory path
+        """
+        self._eps = None
+        if Path(file_name).is_absolute():
+            self.wfile = h5py.File(Path(file_name))
+        else:
+            path = Path(cwd) / Path(file_name)
+            self.wfile = h5py.File(path)
+
+    @property
+    def _n_gk(self):
+        return self.wfile["nGk"][:]
+
+    @property
+    def _fft_idx(self):
+        fft_idx = []
+        off = 0
+        for ngk in self._n_gk:
+            fft_idx.append(self.wfile["fftIdx"][off : off + ngk])
+            off += ngk
+        return fft_idx
```

Comment:
```suggestion
    def _get_fft_idx(self, ik):
        i_start = 0
        if ik > 0:
            i_start = np.sum(self._n_gk[ik - 1])
        i_end = np.sum(self._n_gk[ik])
        return self.wfile["fftIdx"][i_start : i_end])
```

or something like this because only one element of the current list (i.e. `self._fft_idx[ik]` in line 477) is used and appending all the elements is too inefficient